### PR TITLE
change: [M3-8442] - Remove Double border on "Billing & Payment History" table with dark theme.

### PR DESCRIPTION
**Pull request by hasyed-akamai**
## Description 📝
Border for "Billing & Payment History" table was not properly styled for Cloud Manager's dark theme, resulting in double border for the table.

## Changes  🔄
- based on the theme we have changed the border size.

## Preview 📷

| Before | After |
| ------ | ----- |
| Before| After |

### Prerequisites
- Log into accounts which is having any "Billing & Payment History" table entry:

### Reproduction steps
- Landing:
  - Observe the border of "Billing & Payment History" table is reduced.

### Verification steps

- After changes, observe "Billing & Payment History" table border gets half.

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx [hunk 1]
```diff
@@ -374,7 +374,14 @@ export const BillingActivityPanel = React.memo((props: Props) => {
...
     <Grid data-qa-billing-activity-panel xs={12}>
-      <Paper variant="outlined">
+      <Paper
+        sx={(theme) => ({
+          border:
+            theme.name === 'light'
+              ? `1px solid ${theme.borderColors.divider}`
+              : 0,
+        })}
+      >
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx:384]: I don't think this solves the root problem, it just overrides it. I would undo this change and I'll point out what should change.

**Comment by jaalah-akamai** [comment]
Since we're wrapping this entire section in a paper which has a border, we should be able to pass `sx={{ border: 0 }}` to this Table on L441.
https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L441

The other issue on on L479 where that style is no longer relevant and can be removed.
https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L479

Let me know if that fixes all the issues

**Review by cpathipa — COMMENTED** [review]
Agreed with Jaalah's feedback.

File: packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx (hunk 1 above)
Comment (hasyed-akamai) [reply packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx:384]: UpperBorder
The Above Extra Border is removed by removing the L479 where that style is no longer relevant 
https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L479


SideBorder
The Extra Side Border is Removed by using the extra syntax of `sx={{border:0}}` in table section 
https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L441


LowerBorder
The lower Extra Border is able to remove by removing the borderBottom from the table cell component
https://github.com/linode/manager/blob/develop/packages/manager/src/components/TableCell/TableCell.tsx#L34

After Following all the above results are as expected :
Result


but @jaalah-akamaiCould you please let me know if changes to the global component might cause any regression issues ?

**Comment by hasyed-akamai** [comment]
> Since we're wrapping this entire section in a paper which has a border, we should be able to pass `sx={{ border: 0 }}` to this Table on L441. https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L441
> 
> The other issue on on L479 where that style is no longer relevant and can be removed. https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L479
> 
> Let me know if that fixes all the issues

UpperBorder
The Above Extra Border is removed by removing the L479 where that style is no longer relevant 
https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L479


SideBorder
The Extra Side Border is Removed by using the extra syntax of `sx={{border:0}}` in table section 
https://github.com/linode/manager/blob/develop/packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx#L441


LowerBorder
The lower Extra Border is able to remove by removing the borderBottom from the table cell component
https://github.com/linode/manager/blob/develop/packages/manager/src/components/TableCell/TableCell.tsx#L34

After Following all the above results are as expected :
Result


but @jaalah-akamaiCould you please let me know if changes to the global component might cause any regression issues ?

**Comment by jaalah-akamai** [comment]
I think you're on the right approach. The way I would think about this is: We're generating rows via a map. So we're able to determine the last table row. So one way would be to allow an `sx` prop to be applied to the row from which we can target all the cells that have a border bottom.

```
return orderedPaginatedData.map((thisItem, idx) => {
        const lastItem = idx === orderedPaginatedData.length - 1;
        return (
          <ActivityFeedItem
            sxRow={
              lastItem
                ? {
                    '& .MuiTableCell-root': {
                      borderBottom: 0,
                    },
                  }
                : {}
            }
            ... other props
          />
        );
      });
```
Hope that helps... otherwise submit some commits so I can give feedback

**Comment by hasyed-akamai** [comment]
Screenshot 2024-10-23 at 12 59 28 PM

I've followed the steps you mentioned, and everything is working perfectly now! Thank you so much for your help!

**Review by carrillo-erik — APPROVED** [review]
Verified that the most recent changes remove the double border in dark mode, do not cause any unexpected regressions, and resolves the issue rather than override old styles.

**Review by hkhalil-akamai — APPROVED** [review]

**Review by cpathipa — APPROVED** [review]

**Review by pmakode-akamai — APPROVED** [review]

**Pull request merged**